crypto/err/err_local.h

```diff
@@ -76,8 +79,11 @@ static ossl_inline void err_set_data(ERR_STATE *es, size_t i,
     es->err_data_flags[i] = flags;
 }
 
-static ossl_inline void err_clear(ERR_STATE *es, size_t i, int deall)
+static ossl_inline void err_clear(ERR_STATE *es, int i, int deall)
 {
+    if (es->err_buffer[i] == ERR_PACK(ERR_LIB_CRYPTO, 0, ERR_R_MALLOC_FAILURE))
+        es->malloc_failure_reported--;
+    assert(es->malloc_failure_reported >= 0);
```

Comment:
malloc failed, you reported it, and now you're calling assert which could call fprint which could malloc.  This is not good.

Reply:
No - note that this `assert()` is not while reporting an malloc failure, but when removing such an error entry from the queue (e.g., when dumping all errors).

Reply:
Thank you for the explanation. Consider adding a comment.

Reply:
I've added a comment as suggested.